docs/metrics/bleu.md

```diff
@@ -0,0 +1,123 @@
+# BLEU (**B**i**L**ingual **E**valuation **U**nderstudy) Score
+
+The BLEU score is a metric commonly used in a variety of NLP tasks ranging from Machine Translation to Text Summarization, to evaluate the quality of machine-generated text. It quantifies the similarity between the generated text and ground truth text as a score between 0 and 1 — 0 meaning there is no overlap with the ground truth text, and 1 meaning there is a perfect overlap. As the name suggests, it was originally developed for evaluating machine-translation models, but has since been adapted to many different tasks within NLP due to its dynamic nature for measuring textual similarity.
```

Comment:
Would it be correct to think of BLEU as an analog for precision? If so, it would be nice to call this out at the top, something like this:

<img width="808" alt="Screen Shot 2023-09-20 at 8 37 48 AM" src="https://github.com/kolenaIO/kolena/assets/9648886/dee60bd2-cddd-403b-95d1-959ecdafcedc">